### DISCUSSION TITLE
ESP32 and ILI9341 Display not working

### DISCUSSION THREAD

**Discussion by summongod** [arendst/Tasmota#10563] [Q&A] [answered]
I'm having a problem getting the ILI9341 display working with ESP32. The backlight of the screen works but I cannot get displaymode 0,1 or 2 to work and setting 3,4 causes a reset.
To test my hardware I have compiled the Adafruit_ILI9341 graphics test using the same pinouts with the addition of RST on GPIO4. This works as expected and all tests complete.
Could anybody confirm that they have had this screen working with ESP32 on Tasmota 9.2.0.3 or similar and perhaps make some suggestions on what I am doing wrong?

Details of the hardware, pinouts and Tasmota version below.

ESP32 wroom-32 DEVKITV1 (or copy)
Display: 240x320 2.8" SPI TFT LCD Touch Panel Serial Port Module with PCB ILI9341 fa
https://www.ebay.co.uk/itm/164328756356

Module Parameters
GPIO5   -> ILI9341 CS
GPIO12 -> SPI MISO
GPIO13 -> SPI MOSI
GPIO14 -> SPI CLK
GPIO21 -> ILI9341 DC

LED + VCC -> 3.3V
GND          -> GND

Tasmota is built directly from DEV using tasmota32-display with only wifi and password specified in user_config_override.h and upload_port changed in platformio.ini.

Tasmota boot message:
00:00:00.184 SPI: Hardware using GPIO14(CLK), GPIO13(MOSI) and GPIO12(MISO)
00:00:00.482 DSP: ILI9341
00:00:00.485 Project tasmota Tasmota Version 9.2.0.3(display)-1_0_5_RC4(2021-01-14T12:31:40)
-
15:17:49.864 CMD: Display
15:17:49.870 RSL: stat/tasmota_FB24B4/RESULT = {"Display":{"Model":4,"Width":240,"Height":320,"Mode":2,"Dimmer":15,"Size":3,"Font":2,"Rotate":1,"Refresh":2,"Cols":[16,8],"Rows":2}}

**Comment by arendst** [comment, marked as answer]
Just tried with latest development version and it (still) works fine:
```
00:00:00.014 UFS: Type 3 mounted with 52 kB free
00:00:00.042 CFG: Loaded from File, Count 59
00:00:00.056 QPC: Count 1
00:00:00.059 SPI: Hardware using GPIO18(CLK), GPIO23(MOSI) and GPIO19(MISO)
00:00:00.359 DSP: ILI9341
00:00:00.362 SRC: Restart
00:00:00.364 Project tasmota Esp32h Version 9.2.0.3(display)-1_0_5_RC4(2021-01-13T16:03:19)
00:00:00.422 APP: [Version 9.2.0.3(display)]
00:00:00.453 WIF: Attempting connection...
00:00:00.984 WIF: Connecting to AP1 indebuurt_IoT Channel 11 BSSId 18:E8:29:CA:17:C1 in mode 11n as esp32h...
00:00:00.031 APP: [Display mode 1]
00:00:00.969 APP: [Hostname esp32h]
00:00:00.075 WIF: Attempting connection...
00:00:00.976 APP: [Mac 98:F4:AB:6E:58:E8]
00:00:00.074 WIF: Attempting connection...
00:00:03.600 APP: [IP 0.0.0.0]
00:00:03.712 WIF: Connected
00:00:03.970 HTP: Web server active on esp32h with IP address 192.168.2.147
00:00:04.590 RTC: UTC 2021-01-14T16:16:08, DST 2021-03-28T02:00:00, STD 2021-10-31T03:00:00
17:16:08.375 MQT: Attempting connection...
17:16:09.010 MQT: Connected
17:16:09.018 MQT: tele/esp32h/LWT = Online (retained)
17:16:09.024 MQT: cmnd/esp32h/POWER = 
17:16:09.025 MQT: Subscribe to cmnd/esp32h/#
17:16:09.028 MQT: Subscribe to cmnd/tasmotas/#
17:16:09.031 MQT: Subscribe to cmnd/DVES_6E58E8_fb/#
17:16:09.037 MQT: tele/esp32h/INFO1 = {"Module":"ESP32-DevKit","Version":"9.2.0.3(display)","FallbackTopic":"cmnd/DVES_6E58E8_fb/","GroupTopic":"cmnd/tasmotas/"}
17:16:09.047 MQT: tele/esp32h/INFO2 = {"WebServerMode":"Admin","Hostname":"esp32h","IPAddress":"192.168.2.147"}
17:16:09.056 MQT: tele/esp32h/INFO3 = {"RestartReason":"Software reset CPU"}
17:16:09.065 MQT: stat/esp32h/RESULT = {"POWER":"ON"}
17:16:09.070 MQT: stat/esp32h/POWER = ON
17:16:10.567 QPC: Reset
17:16:12.561 APP: Boot Count 19
17:16:13.652 CFG: Saved, Count 60, Bytes 4096
17:16:13.664 MQT: tele/esp32h/STATE = {"Time":"2021-01-14T17:16:13","Uptime":"0T00:00:11","UptimeSec":11,"Heap":111,"SleepMode":"Dynamic","Sleep":10,"LoadAvg":99,"MqttCount":1,"POWER":"ON","Dimmer":83,"Fade":"OFF","Speed":1,"LedTable":"ON","Wifi":{"AP":1,"SSId":"indebuurt_IoT","BSSId":"18:E8:29:CA:17:C1","Channel":11,"RSSI":100,"Signal":-34,"LinkCount":1,"Downtime":"0T00:00:05"}}
```
There is no need to build a version just to change ssid and password; just use the pre-compiled version and add the correct parameters via serial console. 

Connect Rst to 3V3

**Reply by summongod** [reply]
What GPIOs did you use for CS and DC?

**Reply by arendst** [reply]
Same as yours. I have the backlight connected to GPIO04 configured as PWM1 so it can be controlled by a dimmer slider.

**Reply by summongod** [reply]
OK, both my local compile and the standard version now work. It was the RST being set to 3.3V that was the problem. I didn't find anywhere that it was mentioned that it needed to be used. I noticed that the Adafruit library used rst but that the Tasmota version didn't but I didn't look into it. Thanks for taking the time to check and help

**Comment by arendst** [comment]
And response from command display:
```
17:22:37.595 CMD: display
17:22:37.595 SRC: Serial
17:22:37.597 CMD: Group 0, Index 1, Command "DISPLAY", Data ""
17:22:37.606 MQT: stat/esp32h/RESULT = {"Display":{"Model":4,"Width":240,"Height":320,"Mode":1,"Dimmer":2,"Size":1,"Font":1,"Rotate":0,"Refresh":1,"Cols":[40,10],"Rows":16}}
```

**Reply by roesl** [reply]
is it working now ?
I have the same problem

**Comment by Jason2866** [comment]
@roesl Please open a new discussion for your question, where you provide all Infos.
This question is very old and things have changed.
